network/hnswrapper/hnsv2wrapperfake.go

```diff
@@ -7,21 +7,149 @@
 package hnswrapper
 
 import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"reflect"
+	"strings"
+	"sync"
+
 	"github.com/Microsoft/hcsshim/hcn"
 )
 
+const networkName = "azure"
+
+var errorFakeHNS = errors.New("errorFakeHNS Error")
+
+func newErrorFakeHNS(errStr string) error {
+	return fmt.Errorf("%w : %s", errorFakeHNS, errStr)
+}
+
 type Hnsv2wrapperFake struct {
+	Cache FakeHNSCache
+	*sync.Mutex
+}
+
+func NewHnsv2wrapperFake() *Hnsv2wrapperFake {
+	return &Hnsv2wrapperFake{
+		Mutex: &sync.Mutex{},
+		Cache: FakeHNSCache{
+			networks:  map[string]*FakeHostComputeNetwork{},
+			endpoints: map[string]*FakeHostComputeEndpoint{},
+		},
+	}
 }
 
 func (f Hnsv2wrapperFake) CreateNetwork(network *hcn.HostComputeNetwork) (*hcn.HostComputeNetwork, error) {
+	f.Lock()
+	defer f.Unlock()
+
+	f.Cache.networks[network.Name] = NewFakeHostComputeNetwork(network)
 	return network, nil
 }
 
 func (f Hnsv2wrapperFake) DeleteNetwork(network *hcn.HostComputeNetwork) error {
 	return nil
 }
 
-func (Hnsv2wrapperFake) ModifyNetworkSettings(network *hcn.HostComputeNetwork, request *hcn.ModifyNetworkSettingRequest) error {
+func (f Hnsv2wrapperFake) ModifyNetworkSettings(network *hcn.HostComputeNetwork, request *hcn.ModifyNetworkSettingRequest) error {
+	f.Lock()
+	defer f.Unlock()
+	networkCache, ok := f.Cache.networks[network.Name]
+	if !ok {
+		return nil
+	}
+	switch request.RequestType {
+	case hcn.RequestTypeAdd:
+		var setPolSettings hcn.PolicyNetworkRequest
+		err := json.Unmarshal(request.Settings, &setPolSettings)
+		if err != nil {
+			return newErrorFakeHNS(err.Error())
+		}
+		for _, setPolSetting := range setPolSettings.Policies {
+			if setPolSetting.Type == hcn.SetPolicy {
+				var setpol hcn.SetPolicySetting
+				err := json.Unmarshal(setPolSetting.Settings, &setpol)
+				if err != nil {
+					return newErrorFakeHNS(err.Error())
+				}
+				if setpol.PolicyType != hcn.SetPolicyTypeIpSet {
+					// Check Nested SetPolicy members
+					members := strings.Split(setpol.Values, ",")
+					for _, memberID := range members {
+						_, ok := networkCache.Policies[memberID]
+						if !ok {
+							return newErrorFakeHNS(fmt.Sprintf("Member Policy %s not found", memberID))
+						}
+					}
+				}
+				networkCache.Policies[setpol.Id] = &setpol
+			}
+		}
+	case hcn.RequestTypeRemove:
+		var setPolSettings hcn.PolicyNetworkRequest
+		err := json.Unmarshal(request.Settings, &setPolSettings)
+		if err != nil {
+			return newErrorFakeHNS(err.Error())
+		}
+		for _, newPolicy := range setPolSettings.Policies {
+			var setpol hcn.SetPolicySetting
+			err := json.Unmarshal(newPolicy.Settings, &setpol)
+			if err != nil {
+				return newErrorFakeHNS(err.Error())
+			}
+			if _, ok := networkCache.Policies[setpol.Id]; !ok {
+				return newErrorFakeHNS(fmt.Sprintf("[FakeHNS] could not find %s ipset", setpol.Name))
```

Comment:
nitpick: not sure sometime it has `[FakeHNS]` or not. 
It looks overlapped with 
```go
var errorFakeHNS = errors.New("errorFakeHNS Error")
```